week-1-ts-ecosystem/w01e04-light-me-up/src/models/lightBulb.ts

```diff
@@ -0,0 +1,37 @@
+import {PowerSource} from './powerSource';
+
+const ONE_SECOND_IN_MS = 1000;
+
+export class LightBulb {
+  protected readonly powerConsumption = 20
+  #powerSource: PowerSource;
+
+  constructor(powerSource: PowerSource) {
+    this.#powerSource = powerSource;
+  }
+
+  consume() {
+    this.#powerSource.consume(this.powerConsumption);
+  }
+
+  async light(seconds: number) {
```

Comment:
2 rzeczy mi tutaj nie pasują

`async` - po co ?

`seconds` - po co? 
To jest pewien błąd "logiczny". Bo zobacz, każesz żarówce świecić przez `N` sekund. Ale żarówkę się wkręca w dynks... i ona ma swiecić @gkubiak92 . A to że będzie świecić tylko `5` sekund nie wynika z tego że jej "kazaliśmy"
tyle świecić. Tylko z tego że `powerSource` się wyczerpie.

Widzisz różnicę?

A tak to złożenie z `setInterval` - bardzo spoko.

Reply:
Odruchowo dodałem async i o nim zapomniałem :P